src/controllers/trainee/Controller.ts

```diff
@@ -1,81 +1,68 @@
             }
-        })
-    }
-
-    list = (req: Request, res: Response) => {
-        console.log("----------TRAINEE-LIST-----------------");
-        res.send({
-            status: 'OK',
-            message: 'Trainee Listed Successfully',
-            data: [{
-                id: 1,
-                name: ' Ravi Chauhan ',
-                position: ' Trainee Developer ',
-                City: 'Noida'
-            },
-            {
-                id: 2,
-                name: ' Vibhor Garg ',
-                position: ' Trainee Developer ',
-                City: 'Noida'
-            },
-            {
-                id: 3,
-                name: ' Shiva ',
-                position: ' Trainee Developer ',
-                City: 'Delhi'
-            }]
-        })
-    }
-    put = (req: Request, res: Response) => {
-        console.log("----------UPDATE-TRAINEE---------------");
-        res.send({
-            status: 'OK',
-            message: 'Trainee Update Successfully',
-            data: {
-                id: 3,
-                name: ' Ravi Chauhan ',
-                position: ' Trainee Developer ',
-                City: 'Noida'
-            }
-        })
-
-    }
-    delete = (req: Request, res: Response) => {
-        console.log("----------DELETE-TRAINEE---------------");
-        res.send({
-            status: 'OK',
-            message: 'Trainee Delete Successfully',
-            data: {
-                id: 3,
-                name: ' Ravi Chauhan ',
-                position: ' Trainee Developer ',
-                City: 'Noida'
-            }
-        })
-
-    }
+            const pass = await encodedPassword(password);
+            await Object.assign(users, { password: pass });
+            const user = await this.userRepository.create(users)
+            return SystemResponse.success(res, user, 'trainee added sucessfully');
+        } catch (err) {
+            return next({ error: err, message: err });
+        }
+    };
+    list = async (req: Request, res: Response, next: NextFunction) => {
+        try {
+            console.log('-------------INSIDE LIST TRAINEE----------- ');
+            const { skip, limit, sortby } = req.query
+            const countResult = await this.userRepository.count()
+            const user = await this.userRepository.list(Number(skip), Number(limit), sortby)
+            return SystemResponse.success(res, countResult, user, 'Users List');
+        }
+        catch (err) {
+            return next({ error: err, message: err });
+        }
+    };
+    update = async (req: Request, res: Response, next: NextFunction) => {
+        try {
+            console.log('------------INSIDE UPDATE TRAINEE-------------');
+            const { id, dataToUpdate } = req.body;
+            console.log("??????????", id, dataToUpdate);
+            const user = await this.userRepository.update({ _id: id, deletedAt: undefined }, dataToUpdate)
```

Comment:
it should be generic that deletedAt is appended in versioning repository. not by us explicitly.